src/serverlist.cpp

```diff
@@ -321,13 +328,17 @@ void CServerListManager::CentralServerRegisterServer ( const CHostAddress&    In
             // define invalid index used as a flag
             const int ciInvalidIdx = -1;
 
-            // Check if server is already registered. Use address to identify
-            // a server. The very first list entry must not be checked since
+            // Check if server is already registered.
+            // Use external address and local port to identify a server
+            // (there could be more than one server on the external address
+            // but they would have to use separate ports).
+            // The very first list entry must not be checked since
             // this is per definition the central server (i.e., this server)
             int iSelIdx = ciInvalidIdx; // initialize with an illegal value
             for ( int iIdx = 1; iIdx < iCurServerListSize; iIdx++ )
             {
-                if ( ServerList[iIdx].HostAddr == InetAddr )
```

Comment:
The operator== checks that IP number and port are the same. So I see no need to check for equal local port number. Even if we have port translation, each local port of each server gets translated to a different port number so we are fine. So I would remove the "&& ServerList[iIdx].iLocalPortNumber == ServerInfo.iLocalPortNumber". Do you agree?

Reply:
Ah, right, of course.  I think of "addr" only in terms of "IP address" not "IP address and port number".

Reply:
Done
d4636c5 CHostAddr operator== does the job